web/views/api/package_view.ex

```diff
@@ -24,11 +24,24 @@ defmodule HexWeb.API.PackageView do
         release
         |> Map.take(params)
         |> Map.put(:url, api_release_url(Endpoint, :show, package, version))
+        |> if_value(Map.has_key?(release, :retirement), &load_retirement(&1, release.retirement))
       end)
 
     Map.put(entity, :releases, releases)
   end
 
+  defp load_retirement(entity, nil) do
+    Map.put(entity, :retirement, nil)
+  end
```

Comment:
Change the `if_value` above to `if_value(release[:retirement], ...)`. Maybe we can add test so we know for sure? :)